app/webpacker/javascript/cookie_preferences.js

```diff
@@ -81,8 +81,9 @@ export default class CookiePreferences {
 
   clearNonEssentialCookies() {
     Object.keys(Cookies.get()).forEach((key) => {
-      if (!CookiePreferences.functionalCookies.includes(key))
-        Cookies.remove(key);
+      if (!CookiePreferences.functionalCookies.includes(key)) {
+        this.cookieDomains.forEach((d) => Cookies.remove(key, { domain: d }));
```

Comment:
Tiny syntax thing that is quite nice. If you variable name matches the object key, you can use some shorthand.

![image](https://user-images.githubusercontent.com/47089130/144864706-d6f01c1a-7f4a-4f12-8290-d3706d818b53.png)